File: Personagem.py
class Personagem:
    def __init__(self, nome="", vida=100, nivel=1):
        self.nome = nome
        self.vida = vida
        self.nivel = nivel
        self.inventario = []

    # Getters
    def getNome(self):
        return self.nome

    def getVida(self):
        return self.vida

    def getNivel(self):
        return self.nivel

    # Setters
    def setNome(self, nome):
        self.nome = nome

    def setVida(self, vida):
        self.vida = vida

    def setNivel(self, nivel):
        self.nivel = nivel

    # Métodos
    def estaVivo(self):
        if self.vida > 0:
            return True
        else:
            return False

    def atacar(self, dano=10):
        print(f"{self.nome} atacou causando {dano} de dano!")

    def ficha(self):
        print(f"Nome: {self.nome}")
        print(f"Vida: {self.vida}")
        print(f"Nível: {self.nivel}")

        print("\nInventário: ")

        if len(self.inventario) == 0:
            print("Vazio")
        else:
            for item in self.inventario:
                item.descricao()

    def receberDano(self, dano):
        self.vida = self.vida - dano
        return self.vida

    def pegar(self, item):
        self.inventario.append(item)

File: Personagem.java
import java.util.ArrayList;
import java.util.List;

public class Personagem {
    protected  String nome;
    protected  int vida;
    protected  int nivel;
    protected  List<Item> inventario;
    
    public Personagem() {
        nome = "";
        vida = 100;
        nivel = 1;
        inventario = new ArrayList<>();
    }

    public Personagem(String nome, int vida, int nivel) {
        this.nome = nome;
        this.vida = vida;
        this.nivel = nivel;
        this.inventario = new ArrayList<>();
    }


    // Getters
    public String getNome() {
        return nome;
    }

    public int getVida() {
        return vida;
    }

    public int getNivel() {
        return nivel;
    }

    // Setters
    public void setNome(String nome) {
        this.nome = nome;
    }

    public void setVida(int vida) {
        this.vida = vida;
    }

    public void setNivel(int nivel) {
        this.nivel = nivel;
    }

    // Metodos
    boolean estaVivo(){
        if(vida > 0){
            return(true);
        } else{
            return(false);
        }
    }

    public void atacar() {
        System.out.println(nome + " atacou causando 10 de dano!");
    }

    public void atacar(int dano) {
        System.out.println(nome + " atacou causando " + dano + " de dano!");
    }

    public void ficha(){
        System.out.printf("Nome: %s", nome);
        System.out.printf("%nVida: %d", vida);
        System.out.printf("%nNível: %d%n", nivel);

        System.out.printf("%nInventário: ");
        if (inventario.isEmpty()){
            System.err.printf("%nVazio");
        } else {
            for (Item item : inventario){
                item.descricao();
            }
        }
    }

    public int receberDano(int dano){
        return(vida = vida - dano);
    }

    public void pegar(Item item){
        inventario.add(item);
    }
}